lib/FrameManager.js

```diff
@@ -40,6 +40,20 @@ class FrameManager extends EventEmitter {
     this._client.on('Runtime.executionContextCreated', event => this._onExecutionContextCreated(event.context));
   }
 
+  async fetchResourceTree() {
+    const handleTree = tree => {
+      this._onFrameNavigated(tree.frame);
+      if (!tree.childFrames)
+        return;
+
+      for (const child of tree.childFrames)
+        handleTree(child);
+    };
+
+    const { frameTree } = await this._client.send('Page.getResourceTree');
```

Comment:
frameTree should be fetched in `page.create` static method and passed in FrameManager constructor as an argument